Sources/MapboxCoreNavigation/HandlerFactory.swift

```diff
@@ -8,6 +8,7 @@ protocol HandlerData {
     var historyDirectoryURL: URL? { get }
     var targetVersion: String? { get }
     var configFactoryType: ConfigFactory.Type { get }
+    var datasetProfile: ProfileIdentifier { get }
```

Comment:
Having this parameter as a part of caching key has a nuance: since tiles endpoint currently returns the same tiles regardless of our `TilesEndpointConfiguration.dataset` - we are technically don't need to track this parameter. Currently this  may produce "unexpected" re-caching, when user does modify `ProfileIdentifier` in his `RouteOptions` but forgets to do so in `MapboxRoutingProvider` (which may be used even before the actual `NavigationViewController` to request the initial route) or in `TilesetDescriptorFactory`.
In general, this setting now looks like something `NavigationSettings` should take care of, but since it is part of `DirectionsOptions` we have to expose it as an argument. It would help to make such arguments with no default value to highlight the change and force user to setup correct profile, but it is not feasible since we are not modifying public API in patch releases.
Maybe we should consider pushing this feature to the next major release then? /cc @1ec5

Reply:
Tricky situation. I think setting a default value of `automobile` in `MapboxCoreNavigation.Navigator.datasetProfile` seems reasonable for now. If you have ideas about how to simplify this feature across the two libraries, let’s ticket it out and put it on the v3.0 milestone for later consideration.

Reply:
I've captured my thoughts in corresponding discussion about our future Nav Settings with reference to this ticket as an example.